crates/exex/exex/src/wal/cache.rs

```diff
@@ -8,14 +11,14 @@ use reth_primitives::{BlockNumHash, B256};
 ///
 /// This cache is needed to avoid walking the WAL directory every time we want to find a
 /// notification corresponding to a block or a block corresponding to a hash.
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub struct BlockCache {
     /// A mapping of `File ID -> List of Blocks`.
     ///
     /// For each notification written to the WAL, there will be an entry per block written to
     /// the cache with the same file ID. I.e. for each notification, there may be multiple blocks
     /// in the cache.
-    files: BTreeMap<u64, VecDeque<CachedBlock>>,
+    files: Arc<RwLock<BTreeMap<u64, VecDeque<CachedBlock>>>>,
```

Comment:
this will likely be removed, i don't think that we need an ordered mapping anymore, according to https://github.com/paradigmxyz/reth/issues/11202